gateway/src/main/java/org/georchestra/gateway/security/oauth2/OpenIdConnectUserMapper.java

```diff
@@ -144,18 +166,80 @@ public class OpenIdConnectUserMapper extends OAuth2UserMapper {
     }
 
     protected @Override Optional<GeorchestraUser> map(OAuth2AuthenticationToken token) {
+        if (config.createNonExistingUsersInLDAP) {
+            String uid = "oauth2_" + token.getAuthorizedClientRegistrationId() + "_" + token.getName();
+
+            UserMapperImpl mapper = new UserMapperImpl();
+            mapper.setRoleDao(roleDao);
+            List<String> protectedUsers = Collections.emptyList();
+            UserRule rule = new UserRule();
+            rule.setListOfprotectedUsers(protectedUsers.toArray(String[]::new));
+            UsersApiImpl usersApi = new UsersApiImpl();
+            usersApi.setAccountsDao(accountDao);
+            usersApi.setMapper(mapper);
+            usersApi.setUserRule(rule);
+
+            Optional<GeorchestraUser> userOpt = usersApi.findByUsername(uid);
+
+            if (userOpt.isEmpty()) {
+                try {
+                    OidcUser oidcUser = (OidcUser) token.getPrincipal();
+                    String password = RandomStringUtils.randomAlphanumeric(16);
+                    Account newAccount = ExtendedAccountFactory.createBrief(uid, password, oidcUser.getGivenName(),
+                            oidcUser.getFamilyName(), oidcUser.getEmail(), "", "", "");
+                    newAccount.setPending(false);
+                    accountDao.insert(newAccount);
+                    roleDao.addUser(Role.USER, newAccount);
+                    userOpt = usersApi.findByUsername(uid);
+                } catch (DuplicatedUidException e) {
+                } catch (DuplicatedEmailException e) {
+                } catch (DataServiceException e) {
+                }
```

Comment:
to be revisited later on ?

Reply:
Yes, but do you know what to do in case of Exception ? What should happen to the end user ?